### PR TITLE
DBZ-3832 System test-suite ability to prepare OCP environment

### PR DESCRIPTION
**Pull request by MartinMedek**
https://issues.redhat.com/browse/DBZ-3832

### CODE REVIEW

File: debezium-testing/debezium-testing-system/pom.xml [hunk 1]
```diff
@@ -57,10 +59,13 @@
 
     <!--Strimzi configuration-->
     <strimzi.operator.name>strimzi-cluster-operator</strimzi.operator.name>
+    <strimzi.operator.channel>stable</strimzi.operator.channel>
     <strimzi.operator.connectors>true</strimzi.operator.connectors>
     <strimzi.crd.version>v1beta2</strimzi.crd.version>
     <strimzi.kc.build>true</strimzi.kc.build>
     <strimzi.kc.log.level.dbz>DEBUG</strimzi.kc.log.level.dbz>
+    <strimzi.operator.subscription.chanel>stable</strimzi.operator.subscription.chanel>
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/pom.xml:67]: I don't really see difference between strimzi.operator.channel and strimzi.operator.subscription.channel

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/kafka/StrimziOperatorDeployer.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            TimeUnit.SECONDS.sleep(1);
+        }
+
+        throw new IllegalStateException("Kafka operator deployment not created!");
+    }
+
+    private Optional<Deployment> getKafkaOperatorDeployment() {
+        return ocp.resources(Deployment.class).inNamespace(project).list().getItems()
+                .stream()
+                .filter(c -> c.getMetadata().getName().contains("strimzi-cluster-operator") || c.getMetadata().getName().contains("amq-streams-cluster-operator"))
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/kafka/StrimziOperatorDeployer.java:89]: I would prefer to check deployment by label and not name.

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/kafka/builders/FabricKafkaOperatorSubscriptionBuilder.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    private static final String INSTALL_PLAN_APPROVAL = "Automatic";
+
+    protected FabricKafkaOperatorSubscriptionBuilder(SubscriptionBuilder builder) {
+        super(builder);
+    }
+
+    public static FabricKafkaOperatorSubscriptionBuilder base() {
+        SubscriptionBuilder builder = new SubscriptionBuilder().withKind("Subscription")
+                .withApiVersion("operators.coreos.com/v1alpha1");
+        return new FabricKafkaOperatorSubscriptionBuilder(builder);
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/kafka/builders/FabricKafkaOperatorSubscriptionBuilder.java:27]: Remove Fabric from name of builder.

**Review by obabec — CHANGES_REQUESTED** [review]

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/kafka/builders/KafkaSubscriptionBuilder.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        return new KafkaSubscriptionBuilder(builder);
+    }
+
+    public KafkaSubscriptionBuilder withCommunityConfig() {
+        builder.withMetadata(new ObjectMetaBuilder()
+                .withName(STRIMZI_OPERATOR_NAME)
+                .build())
+                .withSpec(new SubscriptionSpecBuilder()
+                        .withName(STRIMZI_OPERATOR_NAME)
+                        .withSource(COMMUNITY_OPERATORS_SOURCE)
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/kafka/builders/KafkaSubscriptionBuilder.java:42]: I would rather go with SubscriptionConstants.COMMUNITY_OPERATORS_SOURCE style and not importing 6 different static variables.

File: debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/fixtures/OcpClient.java [hunk 5]
```diff
@@ -5,24 +5,47 @@
  */
 package io.debezium.testing.system.fixtures;
 
-import static io.debezium.testing.system.tools.OpenShiftUtils.isRunningFromOcp;
+import static io.debezium.testing.system.tools.ConfigProperties.OCP_PROJECT_DB2;
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/fixtures/OcpClient.java:8]: I don't like these single imports. Lets import `ConfigProperties` class only.

**Review by obabec — CHANGES_REQUESTED** [review]

File: debezium-testing/debezium-testing-system/pom.xml [hunk 6]
```diff
@@ -307,6 +311,12 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>org.junit.platform</groupId>
+      <artifactId>junit-platform-launcher</artifactId>
+      <version>${version.junit.platform.launcher}</version>
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/pom.xml:317]: Test scope should be added for this dependency.

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/operatorutil/OpenshiftOperatorEnum.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * Constants for Apicurio and Strimzi operators
+ */
+public enum OpenshiftOperatorEnum {
+    APICURIO("Apicurio",
+            PRODUCT_BUILD ? "service-registry-operator" : "apicurio-registry",
+            ConfigProperties.APICURIO_OPERATOR_CHANNEL,
+            ConfigProperties.OCP_PROJECT_REGISTRY + "-opgroup"),
+    STRIMZI("Strimzi",
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/operatorutil/OpenshiftOperatorEnum.java:20]: Docs...

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/operatorutil/OperatorUtil.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        return operatorEnum.getOperatorChannel();
+    }
+
+    public static void deployOperator(OpenShiftClient ocp, OpenshiftOperatorEnum operatorEnum, String namespace) throws InterruptedException {
+        OpenShiftUtils utils = new OpenShiftUtils(ocp);
+        utils.createOrReplaceOperatorGroup(namespace, operatorEnum.getOperatorGroupName());
+
+        String channel = OperatorUtil.getOperatorChannelName(operatorEnum);
+
+        Subscription subscription;
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/operatorutil/OperatorUtil.java:36]: Docs...

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/registry/builders/ApicurioSubscriptionBuilder.java [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+                .withApiVersion(SubscriptionConstants.SUBSCRIPTION_API_VERSION);
+        return new ApicurioSubscriptionBuilder(builder);
+    }
+
+    @Override
+    public Subscription build() {
+        return builder.build();
+    }
+
+    public ApicurioSubscriptionBuilder withProductConfig() {
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/registry/builders/ApicurioSubscriptionBuilder.java:35]: Docs again

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/operatorutil/OperatorUtil.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+                    .withChannel(channel)
+                    .build();
+        }
+        else {
+            subscription = ApicurioSubscriptionBuilder.base()
+                    .withCommunityConfig()
+                    .withNamespace(namespace)
+                    .withChannel(channel)
+                    .build();
+        }
```
Comment (novotnyJiri) [inline debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/operatorutil/OperatorUtil.java:63]: I would refactor this part of code with 4 if-else branches and instead use some nested ifs with duplicate code extracted

**Review by novotnyJiri — COMMENTED** [review]
I doubt we want the zip files in our upstream. I may be wrong though

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/operatorutil/OperatorUtil.java (hunk 10 above)
Comment (obabec) [reply debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/operatorutil/OperatorUtil.java:63]: Good catch with the zip files I did not catch that.

**Comment by obabec** [comment]
@jcechace Could you check this PR? I have seen it many times and I am sure that I missed something...

**Comment by jcechace** [comment]
LGTM from a high level POV. 

@obabec feel free to merge, I would say that issues will be polished as we go with separate PRs :) 

@MartinMedek good work on this one!

**Review by jcechace — APPROVED** [review]

**Review by obabec — APPROVED** [review]
@MartinMedek please rebase and squash commits and I will merge PR after that :)

**Pull request merged**